TD-3228 Create `PrototypePreview` component

AronPetrauskas opened the pull request:
<!-- Insert YouTrack link if relevant. This should be a clickable link -->
<!-- Pick relevant action word --->

Closes [TD-3228](https://sce.myjetbrains.com/youtrack/issue/TD-3228)
Contributes [TD-3196](https://sce.myjetbrains.com/youtrack/issue/TD-3196)

## Changes

<!-- Let the reviewer know the high-level and detailed changes to look out for -->
<!-- For a bug, this section could instead be bug description & resolution -->

A brief description of what this pull request solves / introduces.

- Creates a new component `PrototypePreview`
- Updates the `StatusIcon` component to expose the padding prop, so this can be adjusted
- Updated the `StatusLabel` component to expose the `gap` prop, so this can be adjusted.
- Updates the tests for `StatusIcon` and `StatusLabel` to test the added functionality
- Adds tests for `PrototypePreview`

Maybe a short description of what is not included and will come in future work where appropriate.

## Dependencies

<!-- Does this branch need to be tested alongside branches from other apps? -->
<!-- Does this branch need to be merged after another Pull Request? -->

## UI/UX

[FIGMA](https://www.figma.com/design/WoHtIQKM6eLlOUuzK1zlpB/VIRTO.TEST?node-id=1789-10490&node-type=canvas&t=KdU30i7WnfrwS13F-0)

Default light:
![image](https://github.com/user-attachments/assets/1376b9e8-ab61-44c7-901f-e784854cbba9)

Default dark:
![image](https://github.com/user-attachments/assets/8f51cc4f-e9ba-4b88-b18e-d94d23b8527c)

Overflow:
![image](https://github.com/user-attachments/assets/88192e6c-51ce-4f1f-b87b-701b4a79ab9f)


<!-- Add in screen grabs / screen recordings of the feature. Tag the UI/UX designer if you require specific feedback / approval -->
<!-- If this PR solves a bug, consider including a before and after so that the reviewer can reproduce and confirm fixed -->

## Testing notes
1. Load up the storybook from the CI/CD GitHub process. Navigate to `PrototypePreview`
2. Make sure the component matches the [FIGMA](https://www.figma.com/design/WoHtIQKM6eLlOUuzK1zlpB/VIRTO.TEST?node-id=1789-10490&node-type=canvas&t=KdU30i7WnfrwS13F-0)
3. Play around with the entries to the component. Make sure that all text overflow is controlled. Long text should truncate and nothing should clash.
4. Make sure clicking on the Prototype name opens a link in a new tab.

<!-- Help the reviewer test your feature with some specific steps, point them towards test data and provide scripts or postman configs etc. -->

## Author checklist

Before I request a review:

<!-- Strikethrough any items that are not relevant to this PR -->

- [x] I have reviewed my own code-diff.
- [x] I have tested the changes in Docker / a deploy-preview.
- [x] I have assigned the PR to myself or an appropriate delegate.
- [x] I have added the relevant labels to the PR.
- [x] I have included appropriate tests.
- [x] I have checked that the Lint and Test workflows pass on Github.
~- [ ] I have populated the deploy-preview with relevant test data.~
- [x] I have tagged a UI/UX designer for review if this PR includes UI/UX changes.

AronPetrauskas commented:
@Sowbhagya-ipg can you please check and confirm that the `PrototypePreview` component works and looks as expected?

Sowbhagya-ipg commented:
Can you reduce the row gap from 16px to 8px?  @AronPetrauskas 
![Uploading Screenshot 2024-12-05 at 1.33.33 pm.png…]()

Sowbhagya-ipg commented:
Can we also remove tooltip on the name and description. @AronPetrauskas

Sowbhagya-ipg commented:
Sorry closed the PR by mistake. 🤦🏻‍♀️

stevenmcsorleyipg reviewed (approved):
LGTM

Sowbhagya-ipg commented:
Looks good @AronPetrauskas

AronPetrauskas commented:
Changes since last review:
- ScenarioPreview was not being exported, fixed that.
- PrototypePreview quality props had to be extended to match the ones it gets in VIRTO
- The Name of all the three preview cards had an anchor tag that was expanding past the text of the name, this was fixed.
- `LinkWithPreview` component was adjusted to allow passing `whiteSpace: "normal"` via sx to allow wrapping.
- Quality prop on `PrototypePreview` was made optional
- Fixed Date and User not aligning centered on `ScenarioPreview`

stevenmcsorleyipg reviewed (approved):
Good demo, LGTM

AronPetrauskas commented:
This PR will remain open until the two following VIRTO PR's are approved 
- https://github.com/IPG-Automotive-UK/virto/pull/1242
- https://github.com/IPG-Automotive-UK/virto/pull/1257

This is so that any UI changes requested as part of the review process can be addressed in this PR.
